.github/workflows/run-tests.yml

```diff
@@ -1,42 +1,52 @@
-name: run-tests
+name: Tests
 
 on:
   push:
+    branches: [master]
   pull_request:
   schedule:
-    - cron: "0 0 * * *"
+    - cron: '0 0 * * *'
 
 jobs:
   test:
-    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
-
     strategy:
-      fail-fast: true
       matrix:
-        php: [7.4, 7.3, 8.0]
+        php: ['7.3', '7.4', '8.0']
         os: [ubuntu-latest, macos-latest]
-        dependency-version: [lowest, highest]
+        dependency-version: [prefer-lowest, prefer-stable]
+        include:
+          - phpunit: '9.*'
+          - php: '7.3'
+            phpunit: '8.*'
 
-    name: P${{ matrix.php }} - ${{ matrix.dependency-version }} - ${{ matrix.os }}
+    name: PHP ${{ matrix.php }}, ${{ matrix.dependency-version }}, ${{ matrix.os }}
 
     runs-on: ${{ matrix.os }}
 
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
 
-      - name: Setup PHP
-        uses: shivammathur/setup-php@v1
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.composer/cache/files
+          key: ${{ runner.os }}-php-${{ matrix.php }}-composer-${{ matrix.dependency-version }}-${{ hashFiles('composer.json') }}
+          restore-keys: |
+            ${{ runner.os }}-php-${{ matrix.php }}-composer-${{ matrix.dependency-version }}-
+            ${{ runner.os }}-php-${{ matrix.php }}-composer-
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
           extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite3, pdo_sqlite
 
-      - uses: ramsey/composer-install@v1
-        id: composer-run
-        continue-on-error: true
-        with:
-          dependency-versions: ${{ matrix.dependency-version }}
+      - name: Install dependencies
```

Comment:
@bakerkretzmar  Why are we installing dependencies here? Is it to test that they can install successfully?

Reply:
We need them installed to run the tests in the next step. The `ramsey/composer-install` action was doing this exact same thing but handling the actual composer commands itself, I changed it back to a custom step so that we have more control over the phpunit and `illuminate/view` versions that get installed (`ramsey/composer-install` only lets you specify the stability).